### ISSUE TITLE
Max pooling incorrectly implemented (?)

### ISSUE THREAD

**Issue by mallamanis** [tejaskannan/r252-deep-code-search#3]
Max pooling seems to be incorrectly implemented with masking. Masks, set elements to `0` instead of `-inf` [here](https://github.com/tejaskannan/r252-deep-code-search/blob/7b04afe6b02fd23ce9057e42ef040a7a005ade67/model.py#L287). This means that in many cases a constant zero element is used as an embedding. The reason is that some non-masked tokens will have embeddings with negative numbers (and that's okay!).

 Instead try
```
 token_emb -= -(1- tf.cast(token_mask, dtype=tf.float32)) * LARGE_NUMBER
```
this should set the masked items to be smaller than all others.

A similar issue seems to exist with the attention mechanism. When you do the softmax to get the attention weights, the masked elements should have very small logits. I suggest again the trick above again.

Also, instead of `tf.layers.maxpooling_1d` and `tf.squeeze` [here](https://github.com/tejaskannan/r252-deep-code-search/blob/7b04afe6b02fd23ce9057e42ef040a7a005ade67/model.py#L380) you can use [`tf.reduce_max()`](https://www.tensorflow.org/versions/r1.11/api_docs/python/tf/reduce_max?hl=en).
